run-build-functions.sh

```diff
@@ -684,6 +684,34 @@ set_go_import_path() {
   fi
 }
 
+prep_functions() {
+  local functionsDir=$1
+  local zisiTempDir=$2
+
+  if [[ -z "$functionsDir" ]]
+    then
+      echo "Skipping functions preparation step: no functions directory set"
+      return 0
+      else
+        echo Function Dir: $functionsDir
+  fi
+  if [[ -z "$zisiTempDir" ]]
+    then
+      echo "Skipping functions preparation step: no temp directory set"
+      else
+        echo TempDir: $zisiTempDir
+  fi
+    echo Prepping functions with zip-it-and-ship-it $(zip-it-and-ship-it --version)
+
+    if zip-it-and-ship-it $functionsDir $zisiTempDir
+    then
+      echo "Prepping functions complete"
+      else
+        echo "Error prepping functions"
+        exit 1
+      fi
```

Comment:
indentation seems a bit hard to read. The most common way to write the conditional is:
```
if ...; then
  foo
else
  bar
fi
```

example:
```
if zip-it-and-ship-it $functionsDir $zisiTempDir; then
    echo "Prepping functions complete"
else
    echo "Error prepping functions"
    exit 1
fi
```

Reply:
Thank you, I'm not great at bash.

Reply:
no one is good at bash ™️

Reply:
OK this should be good now.